Refactor FXIOS-7301 - Remove 2 closure_body_length violations from ImageButtonWithLabel.swift and DefaultBrowserOnboardingView.swift, and decrease threshold

ionixjunior opened the pull request:
## :scroll: Tickets
[Jira ticket](https://mozilla-hub.atlassian.net/browse/FXIOS-7301)
[Github issue](https://github.com/mozilla-mobile/firefox-ios/issues/16442)

## :bulb: Description
This PR removes 2 `closure_body_length` violations from the `ImageButtonWithLabel.swift` file and `DefaultBrowserOnboardingView.swift` file. The second one is from Focus project, but the SwiftLint showed it as a warning when I decreased the threshold. 

This PR extract some views into new properties to decrease the length of the closure. The threshold was defined to 49.

This PR is part of a series of PRs described here https://github.com/mozilla-mobile/firefox-ios/issues/16442#issuecomment-2197676804 and here https://github.com/mozilla-mobile/firefox-ios/issues/16442#issuecomment-2549957143.

## :pencil: Checklist
You have to check all boxes before merging
- [x] Filled in the above information (tickets numbers and description of your work)
- [x] Updated the PR name to follow our [PR naming guidelines](https://github.com/mozilla-mobile/firefox-ios/wiki/Pull-Request-Naming-Guide)
- [ ] Wrote unit tests and/or ensured the tests suite is passing
- [ ] When working on UI, I checked and implemented accessibility (minimum Dynamic Text and VoiceOver)
- [ ] If needed, I updated documentation / comments for complex code and public methods
- [ ] If needed, added a backport comment (example `@Mergifyio backport release/v120`)

FilippoZazzeroni commented on .swiftlint.yml:
```diff
@@ -103,8 +103,8 @@ line_length:
 closure_body_length:
-  warning: 50
-  error: 50
+  warning: 49
```
great 🎉

FilippoZazzeroni reviewed (commented):
LGTM @ionixjunior just as a nit where possible could we add more descriptive names for the subviews variable ?

FilippoZazzeroni commented on focus-ios/BlockzillaPackage/Sources/Onboarding/SwiftUI Onboarding/DefaultBrowserOnboardingView.swift:
```diff
@@ -15,54 +15,20 @@ struct DefaultBrowserOnboardingView: View {
-                    .cornerRadius(.radius)
-            })
-            .padding(.bottom, .skipButtonPadding)
+            topButton
```
for example those two buttons it seems they cover a specific action. maybe they could be something like `defaultSettingBrowserButton`

ionixjunior commented:
Thank you for your suggestion, @FilippoZazzeroni. I've updated the property names to be more descriptive, and rebased the branch. Let me know if there's anything else I can improve.

FilippoZazzeroni commented on focus-ios/BlockzillaPackage/Sources/Onboarding/SwiftUI Onboarding/DefaultBrowserOnboardingView.swift:
```diff
@@ -15,54 +15,20 @@ struct DefaultBrowserOnboardingView: View {
-                }, label: {
-                    Image.close
-                })
+                defaultBrowserCloseButton
```
what do you think this could be directly `closeButton` or `closeOnboardingButton` ?

ionixjunior replied:
I considered that, but when I looked at the enum, I preferred to keep the prefix for clarity. Here’s the enum:

```swift
public enum Action {
    case getStartedAppeared
    case getStartedCloseTapped
    case getStartedButtonTapped
    case defaultBrowserCloseTapped
    case defaultBrowserSettingsTapped
    case defaultBrowserSkip
    case defaultBrowserAppeared
    // Terms Of Service
    case onAcceptAndContinueTapped
    case openTermsOfUse(URL)
    case openPrivacyNotice(URL)
}
```

I really like the idea of using `closeOnboardingButton`, but it might cause some confusion since there are two "close" actions here: `getStartedCloseTapped` and `defaultBrowserCloseTapped`, which is the one we're dealing with.

That said, this argument might not hold if we consider the screen context. It really much simpler to keep the names as `closeOnboardingButton`, `skipOnboardingButton` and `openSettingsButton`. You convinced me. I agree with you. I've made these updates and rebased. 

Thanks for your suggestions!

FilippoZazzeroni replied:
Yes i agree @ionixjunior i think we should consider just the screen context so it's easier to picture the component. 😃

FilippoZazzeroni commented on focus-ios/BlockzillaPackage/Sources/Onboarding/SwiftUI Onboarding/DefaultBrowserOnboardingView.swift:
```diff
@@ -15,54 +15,20 @@ struct DefaultBrowserOnboardingView: View {
-            })
-            .padding(.bottom, .skipButtonPadding)
+            defaultBrowserSettingsButton
+            defaultBrowserSkipButton
```
maybe this could be `skipOnboardingButton` ??

FilippoZazzeroni reviewed (commented):
LGTM @ionixjunior just a little fine tune of the names

FilippoZazzeroni reviewed (approved):
Great work @ionixjunior 🎉
